iron-form.html

```diff
@@ -264,13 +264,16 @@
 
       _saveInitialValues: function() {
         var nodes = this._getSubmittableElements();
+        var props = ['value', 'checked', 'invalid'];
         for (var i = 0; i < nodes.length; i++) {
           var node = nodes[i];
           if (!this._defaults.has(node)) {
-            this._defaults.set(node, {
-              checked: node.checked,
-              value: node.value,
-            });
+            this._defaults.set(node, props.reduce(function(defaults, prop) {
```

Comment:
oh hmm i don't know about setting _all_ the props. What do you think about just doing this for checked/invalid/value. I don't know if resetting an element wrt its form life involves resetting _all_ its properties (think how many properties paper-input has from, all of its behaviours)

Reply:
was kinda thinking saving the props as before, but in the setting case at L387 doing something like
```
if (foo.invalid || defaults.invalid) { foo.invalid = defaults.invalid }
```

or something that isn't garbage code like that

Reply:
it's not setting all the properties, but only the keys present in `props` (see [line 271 above](https://github.com/PolymerElements/iron-form/pull/248/files#diff-c433ce3e372cd3d7c5ced2ce41731fe2R267))

`props.reduce(...` produces an object that will contain at most those 3 keys and at least no keys (empty object).

Reply:
ohhhh i missed `props`

Reply:
After discussing offline, reverted this convoluted code into something which explains better the intent (hopefully). In short: we want to always save `value` as that's what gets submitted anyways, and optionally `checked` (e.g. for native checkboxes) and `invalid` (for backward compat)